crates/misc/run-examples/src/main.rs

```diff
@@ -1,36 +1,74 @@
 use anyhow::Context;
```

Comment:
Is it possible to remove this crate entirely? It seems unfortunate to have both this runner script for CMake plus all the CMake support. I was hoping that this `run-examples` could be entirely deleted in favor of some CMake scripting. I don't know much about CMake myself though.

Reply:
Setting aside the C examples for a second, what would that look like for the Rust examples? Some of the examples have nested Rust subprojects that need to be built first to get a .wasm blob, and I couldn't find a way to make `cargo run --example` trigger that build first. I also don't see a way to ask cargo to run all examples, just a specific one by name.

My guess is that there's still value in this run-examples crate, so I'd like to merge this PR. If we figure out how to replace what it does with a tiny shell script or something, that'd be great as another PR.

Reply:
Cmake would be able to build, it would look similar to the below
```
include(ExternalProject)
ExternalProject_Add(
	wasmtime-crate
	DOWNLOAD_COMMAND ""
	CONFIGURE_COMMAND ""
	INSTALL_COMMAND "${WASMTIME_INSTALL_COMMAND}"
	BUILD_COMMAND cargo build ${WASMTIME_BUILD_TYPE_FLAG}
	BINARY_DIR ${CMAKE_CURRENT_SOURCE_DIR}
	BUILD_ALWAYS ON)
add_library(wasmtime INTERFACE)
add_dependencies(wasmtime wasmtime-crate)
```
And actually it appears you [can test](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html) with cmake, I've never done it but it can be done

Reply:
I had glanced at CTest while I was reviewing this PR earlier and at a very brief glance I didn't think it was meant for this kind of use: it seemed like it was more about running a test suite, with the usual facilities for reporting number of passing/failing tests and so on. I assumed that would be a poor fit, but I could be wrong.

I think it'd be unfortunate to require people to use CMake if they just want to try the Rust examples, so I'm skeptical about building the wasm blobs that way.

Reply:
I've just added integration with ctest

Reply:
Alright, that was simpler than I expected. :laughing: Nice!

I see this version fails on Windows because `-j4` isn't supported there. It's cool that you can run the examples in parallel but I'd also be happy without that optimization.